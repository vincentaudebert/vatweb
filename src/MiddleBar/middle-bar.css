.middle-bar {
  flex-grow: 1;
  display: flex; }

.middle-bar__left-panel {
  margin: 0 !important;
  background-color: #252526;
  display: flex;
  align-items: stretch; }

.middle-bar__right-panel {
  margin: 0 !important;
  width: 100%;
  background-color: #1e1e1e;
  font-family: 'Menlo Regular', Helvetica, sans-serif;
  overflow: scroll; }

.middle-bar__left-panel__title {
  height: 0.75rem;
  padding: 0.625rem 0 0.625rem 0.75rem;
  margin-bottom: 0.625rem;
  font-size: 0.625rem;
  box-shadow: 0px 2px 0px rgba(0, 0, 0, 0.2); }
  .middle-bar__left-panel__title:after {
    clear: both;
    content: ''; }

.middle-bar__left-panel__files {
  font-size: 0.875rem; }

.middle-bar__left-panel__options {
  padding-right: 0.75rem; }

.options__icons {
  margin-left: 0.75rem; }

.middle-bar__left-column {
  background-color: #333333;
  padding: 0.625rem;
  width: 1.875rem;
  font-size: 1.5625rem; }
  .middle-bar__left-column svg {
    margin-bottom: 1rem;
    cursor: pointer; }
    .middle-bar__left-column svg:hover {
      color: #fff; }

.middle-bar__explorer {
  width: 13.75rem; }

.seti-icon {
  width: 0.75rem;
  height: 0.75rem; }

.seti-icon--small {
  padding-left: 0.3125rem;
  padding-right: 0.1875rem; }

.seti-icon--big {
  width: 1.25rem;
  height: 1.25rem;
  vertical-align: middle; }

.folder--in {
  padding-left: 1.5rem !important; }

.file {
  padding-left: 0.75rem; }
  .file:hover {
    background-color: #3f3f46; }

.folder {
  padding-left: 0.75rem; }

.tabs {
  position: sticky;
  top: 0;
  margin: 0;
  width: 100%;
  background-color: #252526;
  box-shadow: 0px 2px 0px rgba(0, 0, 0, 0.2); }

.tab {
  padding: 0.25rem 0.75rem 0.25rem 0.625rem;
  border-right: 1px solid #1e1e1e;
  background-color: #2d2d2d; }
  .tab svg {
    color: #2d2d2d; }
  .tab:hover svg {
    color: #fff; }

.tab--active {
  color: #fff;
  background-color: #1e1e1e; }
  .tab--active svg {
    color: #fff; }

.file-content {
  overflow-y: scroll; }
  .file-content pre {
    padding-top: 0.125rem !important; }
  .file-content code:first-child {
    text-align: right !important; }

.explorer--active {
  background-color: #094771;
  color: #fff; }

.file--link {
  text-decoration: none;
  color: inherit;
  display: block; }
  .file--link:hover, .file--link:focus, .file--link:active {
    text-decoration: none;
    color: inherit; }
