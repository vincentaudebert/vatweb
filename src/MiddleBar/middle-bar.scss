@import '../css/variables';
@import '../css/mixins';

.middle-bar {
  flex-grow: 1;
  display: flex;
}

.middle-bar__left-panel {
  margin: 0 !important;
  background-color: $color-grey;
  display: flex;
  align-items: stretch;
}

.middle-bar__right-panel {
  margin: 0 !important;
  width: 100%;
  background-color: $color-dark-grey;
  font-family: 'Menlo Regular', Helvetica, sans-serif;
  overflow: scroll;
}

.middle-bar__left-panel__title {
  height: rem(12px);
  padding: rem(10px) 0 rem(10px) $spacing;
  margin-bottom: rem(10px);
  font-size: rem(10px);
  box-shadow: 0px 2px 0px rgba(0, 0, 0, 0.2);

  &:after {
    clear: both;
    content: '';
  }
}

.middle-bar__left-panel__files {
  font-size: rem(14px);
}

.middle-bar__left-panel__options {
  padding-right: $spacing;
}

.options__icons {
  margin-left: $spacing;
}

.middle-bar__left-column {
  background-color: $color-medium-dark-grey;
  padding: rem(10px);
  width: rem(30px);
  font-size: rem(25px);

  svg {
    margin-bottom: rem(16px);
    cursor: pointer;

    &:hover {
      color: $color-white;
    }
  }
}

.middle-bar__explorer {
  width: rem(220px);
}

.seti-icon {
  width: rem(12px);
  height: rem(12px);
}

.seti-icon--small {
  padding-left: rem(5px);
  padding-right: rem(3px);
}

.seti-icon--big {
  width: rem(20px);
  height: rem(20px);
  vertical-align: middle;
}

.folder--in {
  padding-left: $spacing * 2 !important;
}

.file {
  padding-left: $spacing;

  &:hover {
    background-color: $color-hover-grey;
  }
}

.folder {
  padding-left: $spacing;
}

.tabs {
  position: sticky;
  top: 0;
  margin: 0;
  width: 100%;
  background-color: $color-grey;
  box-shadow: 0px 2px 0px rgba(0, 0, 0, 0.2);
}

.tab {
  padding: rem(4px) rem(12px) rem(4px) rem(10px);
  border-right: 1px solid $color-dark-grey;
  background-color: $color-inactive-grey;

  svg {
    color: $color-inactive-grey;
  }

  &:hover {
    svg {
      color: $color-white;
    }
  }
}

.tab--active {
  color: $color-white;
  background-color: $color-dark-grey;

  svg {
    color: $color-white;
  }
}

.file-content {
  overflow-y: scroll;

  pre {
    padding-top: rem(2px) !important;
  }

  code:first-child {
    text-align: right !important;
  }
}

.explorer--active {
  background-color: $color-selected-blue;
  color: $color-white;
}

.file--link {
  text-decoration: none;
  color: inherit;
  display: block;

  &:hover,
  &:focus,
  &:active {
    text-decoration: none;
    color: inherit;
  }
}
