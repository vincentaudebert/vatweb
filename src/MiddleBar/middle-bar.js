import React from 'react';
import './middle-bar.css';
import Explorer from './Explorer/explorer.container';
import FileContent from './FileContent/file-content.container';

export const MiddleBar = props => {
  return (
    <div className="middle-bar row">
      <div className="left middle-bar__left-panel">
        <Explorer />
      </div>
      <div className="middle-bar__right-panel">
        <FileContent />
      </div>
    </div>
  );
};

export default MiddleBar;
