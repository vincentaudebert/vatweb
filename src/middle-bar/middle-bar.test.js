import React from 'react';
import renderer from 'react-test-renderer';
import MiddleBar from './middle-bar';

jest.mock('./explorer/explorer.container', () => 'div');
jest.mock('./file-content/file-content.container', () => 'div');

describe('MiddleBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MiddleBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
