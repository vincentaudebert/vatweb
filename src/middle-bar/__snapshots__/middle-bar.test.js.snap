// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiddleBar renders correctly 1`] = `
<div
  className="middle-bar row"
>
  <div
    className="left middle-bar__left-panel"
  >
    <div />
  </div>
  <div
    className="middle-bar__right-panel"
  >
    <div />
  </div>
</div>
`;
